neurom/core/_neuron.py

```diff
@@ -28,16 +28,176 @@
+            else:
+                children.pop()
+                yield cur_node
+
+    def iupstream(self):
+        """Iterate from a tree node to the root nodes."""
+        t = self
+        while t is not None:
+            yield t
+            t = t.parent
+
+    def ileaf(self):
+        """Iterator to all leaves of a tree."""
+        return filter(Section.is_leaf, self.ipreorder())
+
+    def iforking_point(self, iter_mode=ipreorder):
+        """Iterator to forking points.
+
+        Args:
+            iter_mode: iteration mode. Default: ipreorder.
+        """
+        return filter(Section.is_forking_point, iter_mode(self))
+
+    def ibifurcation_point(self, iter_mode=ipreorder):
+        """Iterator to bifurcation points.
+
+        Args:
+            iter_mode: iteration mode. Default: ipreorder.
+        """
+        return filter(Section.is_bifurcation_point, iter_mode(self))
+
+    def __eq__(self, other):
+        """Equal when its morphio section is equal."""
+        return self.morphio_section == other.morphio_section
+
+    def __hash__(self):
+        """Hash of its id."""
+        return hash(self.id)
+
+    def __nonzero__(self):
+        """If has children."""
+        return bool(self.children)
+
+    __bool__ = __nonzero__
+
+    @property
+    def points(self):
+        """Returns the section list of points the NeuroM way (points + radius)."""
+        return np.concatenate((self.morphio_section.points,
+                               self.morphio_section.diameters[:, np.newaxis] / 2.),
+                              axis=1)
+
+    @points.setter
+    def points(self, value):
+        """Set the points."""
+        self.morphio_section.points = np.copy(value[:, COLS.XYZ])
+        self.morphio_section.diameters = np.copy(value[:, COLS.R]) * 2
+
+    @property
+    def type(self):
+        """Returns the section type."""
+        return NeuriteType(int(self.morphio_section.type))
+
+    @property
+    def length(self):
+        """Return the path length of this section."""
+        return morphmath.section_length(self.points)
+
+    @property
+    def area(self):
+        """Return the surface area of this section.
+
+        The area is calculated from the segments, as defined by this
+        section's points
+        """
+        return sum(morphmath.segment_area(s) for s in iter_segments(self))
+
+    @property
+    def volume(self):
+        """Return the volume of this section.
+
+        The volume is calculated from the segments, as defined by this
+        section's points
+        """
+        return sum(morphmath.segment_volume(s) for s in iter_segments(self))
+
+    def __str__(self):
+        """Text representation."""
+        parent_id = None if self.parent is None else self.parent.id
+        return (f'Section(id={self.id}, type={self.type}, n_points={len(self.points)})'
+                f'<parent: Section(id={parent_id}), nchildren: {len(self.children)}>')
+
+    __repr__ = __str__
```

Comment:
if you define `__repr__` instead of `__str__` the   `__str__ == __repr__ ` is done automatically

Reply:
done